src/sources/http_scrape/scrape.rs

```diff
@@ -0,0 +1,247 @@
+    /// appended to any parameters manually provided in the `endpoint` option.
+    #[serde(default)]
+    query: HashMap<String, Vec<String>>,
+
+    /// Decoder to use on the HTTP responses.
+    #[configurable(derived)]
+    #[serde(default = "default_decoding")]
+    decoding: DeserializerConfig,
+
+    /// Framing to use in the decoding.
+    #[configurable(derived)]
+    #[serde(default = "default_framing_message_based")]
+    framing: FramingConfig,
+
+    /// Headers to apply to the HTTP requests.
+    /// One or more values for the same header can be provided.
+    #[serde(default)]
+    headers: HashMap<String, Vec<String>>,
+
+    /// TLS configuration.
+    #[configurable(derived)]
+    tls: Option<TlsConfig>,
+
+    /// HTTP Authentication.
+    #[configurable(derived)]
+    auth: Option<Auth>,
+}
+
+impl Default for HttpScrapeConfig {
+    fn default() -> Self {
+        Self {
+            endpoint: "http://localhost:9898/logs".to_string(),
+            query: HashMap::new(),
+            scrape_interval_secs: default_scrape_interval_secs(),
+            decoding: default_decoding(),
+            framing: default_framing_message_based(),
+            headers: HashMap::new(),
+            tls: None,
+            auth: None,
+        }
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+impl HttpScrapeConfig {
+    pub const fn new(
+        endpoint: String,
+        scrape_interval_secs: u64,
+        query: HashMap<String, Vec<String>>,
+        decoding: DeserializerConfig,
+        framing: FramingConfig,
+        headers: HashMap<String, Vec<String>>,
+        tls: Option<TlsConfig>,
+        auth: Option<Auth>,
+    ) -> Self {
+        Self {
+            endpoint,
+            scrape_interval_secs,
+            query,
+            decoding,
+            framing,
+            headers,
+            tls,
+            auth,
+        }
+    }
+}
+
+inventory::submit! {
+    SourceDescription::new::<HttpScrapeConfig>(NAME)
+}
+
+impl_generate_config_from_default!(HttpScrapeConfig);
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "http_scrape")]
+impl SourceConfig for HttpScrapeConfig {
+    async fn build(&self, cx: SourceContext) -> Result<sources::Source> {
+        // build the url
+        let endpoints = vec![self.endpoint.clone()];
+        let urls = endpoints
+            .iter()
+            .map(|s| s.parse::<Uri>().context(sources::UriParseSnafu))
+            .map(|r| r.map(|uri| get_url(&uri, &self.query)))
+            .collect::<std::result::Result<Vec<Uri>, sources::BuildError>>()?;
+
+        let tls = TlsSettings::from_options(&self.tls)?;
+
+        // build the decoder
+        let decoder = DecodingConfig::new(
+            self.framing.clone(),
+            self.decoding.clone(),
+            LogNamespace::Vector,
+        )
+        .build();
+
+        let content_type = self.decoding.content_type(&self.framing).to_string();
```

Comment:
Hi! First of all, a BIG thank you for working on this source. We have been waiting for this source for a while!

I think this should be overridable by configuration and only use `self.decoding.content_type(&self.framing)` as default if no specific content-type was configured by the user for the source.

Unfortunately there is a non-neglectable number of servers that do not respect/handle the `Accept` header correctly, or that only accept content-types not covered by the defined heuristic. Sometimes it is also needed to send `Accept: */*`.

Moreover, it is always possible to do custom decoding in VRL, e.g. `parse_json()` (this is our most common use case). In this case, the source would be configured with the `bytes` decoder, which would be sending `Accept: text/plain` according to the defined heuristic. This would be incorrect as we _still_ want `Accept: application/json` to be sent, just not decode at the source level.

If I understood the content type heuristic correctly, the following content-types are used in `Accept` according to codec/framer used:
Codec | Framer | Content-Type
:--|:--|:--
JSON, NativeJSON | NewlineDelimited | `application/x-ndjson`
GELF, JSON, NativeJSON | CharacterDelimited (by `,`) | `application/json`
Native | Any | `application/octet-stream`
JSON, NativeJSON, Bytes, GELF | Any | `text/plain`
Syslog | Any | `text/plain`

I hope you can take this feedback in consideration :)

Reply:
Oh, I just realised that the user-supplied headers are applied _after_ setting the `Accept` header by the source.
If the user-supplied headers can fully override `Accept` without _appending_ to it, then I guess my previous comment can be dismissed :)

Reply:
Thanks for taking a look at the changes and for the additional context!
I've made a small change to only apply the default (codec based) `Accept` header if the user did not specify one.

Reply:
Thanks for the change, I think it makes 100% sense now!